internal/oci/runtime_oci.go

```diff
@@ -875,14 +875,18 @@ func (r *runtimeOCI) AttachContainer(c *Container, inputStream io.Reader, output
 
 	stdinDone := make(chan error)
 	go func() {
-		var err error
+		var err, closeErr error
 		if inputStream != nil {
 			_, err = utils.CopyDetachable(conn, inputStream, nil)
-			if closeErr := conn.CloseWrite(); closeErr != nil {
-				stdinDone <- closeErr
-			}
+			closeErr = conn.CloseWrite()
+		}
+		if err != nil {
+			stdinDone <- err
+		} else if closeErr != nil {
+			stdinDone <- closeErr
+		} else {
```

Comment:
just thought of this: should probably have a comment here, or maybe instead of sending `stdinDone <- err` it would be more clear to have it be `stdinDone <- nil`

Reply:
@haircommander 
Please take another look.